dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/srt/SRTRequestContext.java

```diff
@@ -132,7 +133,31 @@ public void finish()
                     HttpSession s = (HttpSession)webappToSessionMap.get(wa);
                     IHttpSessionContext ctx =wa.getSessionContext();
                     if (ctx!=null){
-                    	wa.getSessionContext().sessionPostInvoke(s);
+                        // set app classloader as TCCL; JSF might need to load resources during session cleanup
+                        ClassLoader origClassLoader = ThreadContextHelper.getContextClassLoader();
+                        final ClassLoader warClassLoader = this.request.getServletContext().getClassLoader();
+                        if (warClassLoader != origClassLoader) {
+                            if (logger.isLoggable(Level.FINE)) {
+                                if(origClassLoader != null) {
+                                    logger.logp(Level.FINE, CLASS_NAME, "finish", "re-set class loader from --> " + origClassLoader.toString()
+                                                + " ,to --> " + warClassLoader.toString());
+                                }
+                                else {
+                                    logger.logp(Level.FINE, CLASS_NAME, "finish", "re-set class loader from origClassLoader--> null ,to --> " + warClassLoader.toString()); 
+                                }
+                            }
+                            ThreadContextHelper.setClassLoader(warClassLoader);
+                        } else {
+                            origClassLoader = null;
+                        }
+
+                        wa.getSessionContext().sessionPostInvoke(s);
+                            
+                        // reset TCCL to its original loader
+                        if (origClassLoader != null) {
+                            final ClassLoader fOriginalClassloader = origClassLoader;
+                            ThreadContextHelper.setClassLoader(fOriginalClassloader);
+                        }
```

Comment:
I would recommend reset the original classloader in a finally block so that it will be reset in the event that `wa.getSessionContext().sessionPostInvoke(s)` throws an exception.

Reply:
I've added the reset into a `finally` in a new commit